app/com/linkedin/drelephant/spark/heuristics/ConfigurationHeuristic.scala

```diff
@@ -130,29 +151,70 @@ object ConfigurationHeuristic {
     lazy val executorCores: Option[Int] =
       Try(getProperty(SPARK_EXECUTOR_CORES_KEY).map(_.toInt)).getOrElse(None)
 
-    lazy val applicationDuration : Long = {
+    lazy val driverCores: Option[Int] =
+      Try(getProperty(SPARK_DRIVER_CORES_KEY).map(_.toInt)).getOrElse(None)
+
+    lazy val dynamicMinExecutors: Option[Int] =
+      Try(getProperty(SPARK_DYNAMIC_ALLOCATION_MIN_EXECUTORS).map(_.toInt)).getOrElse(None)
+
+    lazy val dynamicMaxExecutors: Option[Int] =
+      Try(getProperty(SPARK_DYNAMIC_ALLOCATION_MAX_EXECUTORS).map(_.toInt)).getOrElse(None)
+
+    lazy val applicationDuration: Long = {
       require(data.applicationInfo.attempts.nonEmpty)
       val lastApplicationAttemptInfo = data.applicationInfo.attempts.last
       (lastApplicationAttemptInfo.endTime.getTime - lastApplicationAttemptInfo.startTime.getTime) / Statistics.SECOND_IN_MS
     }
 
+    lazy val sparkYarnJars: String = getProperty(SPARK_YARN_JARS).getOrElse("")
+
+    lazy val jarsSeverity: Severity = if (sparkYarnJars.contains("*")) {
+        Severity.CRITICAL
+      } else {
+        Severity.NONE
+      }
+
     lazy val serializer: Option[String] = getProperty(SPARK_SERIALIZER_KEY)
 
     /**
-     * If the serializer is either not configured or not equal to KryoSerializer, then the severity will be moderate.
-     */
+      * If the serializer is either not configured or not equal to KryoSerializer, then the severity will be moderate.
+      */
 
     lazy val serializerSeverity: Severity = serializer match {
       case None => Severity.MODERATE
       case Some(`serializerIfNonNullRecommendation`) => Severity.NONE
       case Some(_) => DEFAULT_SERIALIZER_IF_NON_NULL_SEVERITY_IF_RECOMMENDATION_UNMET
     }
 
+    //The following thresholds are for checking if the memory and cores values (executor and driver) are above normal. These thresholds are experimental, and may change in the future.
+    val DEFAULT_SPARK_MEMORY_THRESHOLDS =
+      SeverityThresholds(low = MemoryFormatUtils.stringToBytes("10G"), MemoryFormatUtils.stringToBytes("15G"), severe = MemoryFormatUtils.stringToBytes("20G"), critical = MemoryFormatUtils.stringToBytes("25G"), ascending = true)
+    val DEFAULT_SPARK_CORES_THRESHOLDS =
+      SeverityThresholds(low = 4, moderate = 6, severe = 8, critical = 10, ascending = true)
+
+    val severityExecutorMemory = DEFAULT_SPARK_MEMORY_THRESHOLDS.severityOf(executorMemoryBytes.getOrElse(0).asInstanceOf[Number].longValue)
```

Comment:
Just updated the JIRA ticket LIHADOOP-32520 with the following:

Please also add checks to:

- Make sure the user does not try to turn off dynamic allocation (spark.dynamicAllocation.enabled should be set to true), and it should be a severe error if not.
- spark.dynamicAllocation.executorIdleTimeout should be set to the cluster default value of 150, and it should be a severe error if it is a different value.

Sorry for the last minute addition -- would you be able to make the changes?